### PR TITLE
Add client mock

### PR DESCRIPTION
**Pull request by cw9**
@xichen2020 @robskillington

### CODE REVIEW

**Comment by cw9** [comment]
Wondering if client.go should be moved to client/ so the mock does not stay on top level

File: client.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ package cluster
 
 import (
 	"github.com/m3db/m3cluster/kv"
-	"github.com/m3db/m3x/watch"
+	xwatch "github.com/m3db/m3x/watch"
```
Comment (xichen2020) [inline client.go]: Don't think you need this since the package name should already be xwatch

File: generated/mocks/generate.go [hunk 2]
```diff
@@ -19,5 +19,6 @@
 // THE SOFTWARE.
 
 //go:generate sh -c "mockgen -package=kv -destination=$GOPATH/src/$PACKAGE/kv/mock_store.go -source=$GOPATH/src/$PACKAGE/kv/store.go"
+//go:generate sh -c "mockgen -package=cluster -destination=$GOPATH/src/$PACKAGE/mock_client.go -source=$GOPATH/src/$PACKAGE/client.go"
```
Comment (xichen2020) [inline generated/mocks/generate.go]: Not for this diff, but eventually this should be a build target

**Comment by xichen2020** [comment]
Yeah it makes sense to move `client.go` to its own package

File: generated/mocks/generate.go (hunk 2 above)
Comment (xichen2020) [inline generated/mocks/generate.go]: Also probably change the package name to `client` if you move `client.go` to its own package

File: client.go (hunk 1 above)
Comment (cw9) [inline client.go]: that way the mockgen would fail, it could not find the xwatch package, only works with rename somehow.

File: client/client.go [hunk 3]
```diff
@@ -18,11 +18,11 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-package cluster
+package client
 
 import (
 	"github.com/m3db/m3cluster/kv"
-	"github.com/m3db/m3x/watch"
+	xwatch "github.com/m3db/m3x/watch"
```
Comment (xichen2020) [inline client/client.go]: can we get rid of the alias here? the package name in m3x/watch is already xwatch

**Comment by xichen2020** [comment]
LGTM

**Pull request merged**